java/trace/src/test/scala/org/typelevel/otel4s/java/trace/TracerSuite.scala

```diff
@@ -70,6 +71,25 @@ class TracerSuite extends CatsEffectSuite {
     )
   }
 
+  test("set attributes only once") {
+    val key = "string-attribute"
+    val value = "value"
+    val attribute = Attribute(AttributeKey.string(key), value)
+
+    for {
+      sdk <- makeSdk()
+      tracer <- sdk.provider.tracer("tracer").get
+      _ <- tracer.span("span", attribute).use_
+      spans <- sdk.finishedSpans
+    } yield {
+      assertEquals(spans.map(_.getAttributes.size), List(1))
+      assertEquals(
```

Comment:
Initially, I tried 
```scala
assertEquals(spans.map(_.getAttributes), List(Attributes.builder().put(key, value).build()))
```

but this asserting is failing:
```diff
List(
-  AttributesMap{data={string-attribute=value}, capacity=128, totalAddedValues=1}
+  {string-attribute="value"}
)
```

Seems like an SDK bug.

Reply:
https://github.com/open-telemetry/opentelemetry-java/issues/5148

Reply:
I might have this wrong, because I'm browsing GitHub while a dog barks.  But `ArrayBackedAttributes` inherits [this equality](https://github.com/open-telemetry/opentelemetry-java/blob/59043802e67f784550cd906fe19d8f7521d253f3/api/all/src/main/java/io/opentelemetry/api/internal/ImmutableKeyValuePairs.java#L237), and `AttributesMap` inherits `java.util.HashMap`'s.  I imagine both could be retrofitted with a compatible implementation, but it might be hard to make a lawful implementation with respect to symmetry and other java.util.HashMaps.  Overriding a concrete equals usually causes trouble in Java.

Reply:
Yeah, than `.getAttributes.get(JAttributeKey.stringKey(key))` is enough